src/clients/logging.js

```diff
@@ -0,0 +1,120 @@
+/* Client request logger.
+ */
+import { assign, get } from 'lodash';
+import {
+    extractLoggingProperties,
+} from '@globality/nodule-logging';
+import { getContainer } from '@globality/nodule-config';
+
+export function calculateExecuteTime(executeStartTime) {
+    const executeTime = process.hrtime(executeStartTime);
+    return (executeTime[0] * 1e3) + (executeTime[1] * 1e-6);
+}
+
+export function getElapsedTime(req) {
+    if (!req._startAt) { // eslint-disable-line no-underscore-dangle
```

Comment:
One subtlety here:
 - We treat `nodule-graphql` as a *framework*, meaning it explicitly combines several other libraries
 - We treat `nodule-openapi` as a *library*, meaning it should it should do one thing well

The reference to `req_.startAt` represents coupling to `nodule-logging`. The best practice here would be to do at least one of:
 a) Keep the interdependencies at the framework layer
 b) Build an abstraction here that avoids the dependency
 c) Document this well